scripts/gen_airbyte_classes.py

```diff
@@ -0,0 +1,579 @@
+                            or f"{field.get('title')}Entry"
+                        )
+                        class_definitions = {
+                            **class_definitions,
+                            **get_class_definitions(title, items_data),
+                        }
+                        fields[field_name] = ListType(RawType(title))
+                    else:
+                        fields[field_name] = ListType(RawType(array_type))
+                else:
+                    fields[field_name] = RawType(field_type, const_value=field.get("const"))
+                if field_name not in required_fields:
+                    fields[field_name] = OptType(fields[field_name])
+
+    class_definitions[name] = fields
+    return class_definitions
+
+
+CLASS_TEMPLATE = """
+class {cls_name}:
+    def __init__(self, {fields_in}):
+{self_fields}
+"""
+
+
+SOURCE_TEMPLATE = '''
+class {cls_name}(GeneratedAirbyteSource): {nested_defs}
+    def __init__(self, name: str, {fields_in}):
+        """
+        Airbyte Source for {human_readable_name}
+
+        Documentation can be found at {docs_url}
+        """
+{self_fields}
+        super().__init__("{human_readable_name}", name)
+'''
+
+DESTINATION_TEMPLATE = '''
+class {cls_name}(GeneratedAirbyteDestination): {nested_defs}
+    def __init__(self, name: str, {fields_in}):
+        """
+        Airbyte Destination for {human_readable_name}
+
+        Documentation can be found at {docs_url}
+        """
+{self_fields}
+        super().__init__("{human_readable_name}", name)
+'''
+
+
+def create_nested_class_definition(
+    base_cls_name: str,
+    cls_name: str,
+    cls_def: Dict[str, SchemaType],
+):
+    nested_defs = ""
+    fields_in = ", ".join(
+        [
+            f"{field_name}: {field_type.annotation(scope=base_cls_name, quote=True)}"
+            for field_name, field_type in sorted(
+                cls_def.items(), key=lambda x: isinstance(x[1], OptType)
+            )
+            if field_type.const_value is None
+        ]
+    )
+
+    self_fields = "\n".join(
+        [
+            f'        self.{field_name} = "{field_type.const_value}"'
+            for field_name, field_type in cls_def.items()
+            if field_type.const_value is not None
+        ]
+        + [
+            f"        self.{field_name} = {field_type.get_check(field_name, scope=base_cls_name)}"
+            for field_name, field_type in cls_def.items()
+            if field_type.const_value is None
+        ]
+    )
+    return CLASS_TEMPLATE.format(
+        cls_name=cls_name,
+        fields_in=fields_in,
+        self_fields=self_fields,
+        nested_defs=nested_defs,
+    )
+
+
+def create_connector_class_definition(
+    connector_name_human_readable: str,
+    cls_name: str,
+    cls_def: Dict[str, SchemaType],
+    nested: Optional[List[str]],
+    is_source: bool,
+    docs_url: str,
+):
+    nested_defs = ""
+    if nested:
+        nested_defs = "\n".join(
```

Comment:
https://docs.python.org/3/library/textwrap.html#textwrap.indent might be helpful here